### PR TITLE
Assorted cleanup

### PR DESCRIPTION
**Pull request by apasel422**
r? @Gankro

### CODE REVIEW

**Comment by apasel422** [comment]
Ping.

File: src/lib.rs [hunk 1]
```diff
@@ -54,9 +54,9 @@ impl<T> Node<T> {
     /// Makes the given node come after this one, appropriately setting all other links.
     /// Assuming that self has a `next`.
     #[inline]
-    fn splice_next(&mut self, mut next: Box<Node<T>>) {
+    fn splice_next(&mut self, mut next: Box<Self>) {
         let mut old_next = self.next.take();
-        old_next.as_mut().map(|node| node.prev = Raw::some(&mut *next));
+        if let Some(ref mut node) = old_next { node.prev = Raw::some(&mut *next); }
```
Comment (Gankro) [inline src/lib.rs]: I'm not really convinced this is an improvement. I get some people are objectively opposed to side-effecting `map`, but I really hate if-let for most cases.

**Comment by Gankro** [comment]
Looks basically good modulo my one style nit (maybe just give up on the one-liner if you must have the if-let?)

BTW, I'm not sure if the news ever got to you, because you're like... not on things... but I start at Apple on the 13th, at which point I'm contractually barred from doing... anything that isn't for Apple. So... yeah. I'm basically a goner?

**Comment by apasel422** [comment]
@Gankro Fixed.

Congratulations on the new job! Good luck, and thanks for all the help!

**Pull request merged**